src/+datools/+statistical/+ensemble/RHF.m

```diff
@@ -1,158 +1,167 @@
-            
+
             % Observable
             Hxf = obj.Observation.observeWithoutError(tc, xf);
+            % mean of observable
             Hxfm = mean(Hxf, 2);
-            
-            ind1 = 1:ensN-1; ind2 = 2:ensN;
-            
-            % do for each state variable
-            for i = 1:size(xf,1)
-                [xfs, sortin] = sort(xf(i,:));
-                
-                % try inflating the background
-                xfs = mean(xfs) + inflation*(xfs - mean(xfs));
-                
-                prior_ht = zeros(1, ensN + 1);
+
+            % will be used for slicing
+            ind1 = 1:ensN - 1;
+            ind2 = 2:ensN;
+
+            % assimilate each state variable
+            for i = 1:size(xf, 1)
+                % sort the forecast ensemble
+                %xfs = sorted forecast ensemble
+                [xfs, sortin] = sort(xf(i, :));
+
+                priorht = zeros(1, ensN+1);
                 % find the height of uniform part
                 for ii = 2:ensN
-                    prior_ht(ii) = 1/ ((ensN + 1)*abs((xfs(ii) - xfs(ii-1))));
+                    priorht(ii) = 1 / ((ensN + 1) * abs((xfs(ii) - xfs(ii-1))));
                 end
-                
+
                 Var = xfv(i);
-                
+
                 if strcmp(obj.Tail, 'Gaussian') == 1
-                    muleft = xfs(1) - sqrt(2*Var)*erfinv(2/(ensN + 1) -1);
-                    muright = xfs(end) - sqrt(2*Var)*erfinv(1 - 2/(ensN + 1));
-                    mu = [muleft muright];
+                    % find the mean of the left gaussian tail
+                    muleft = xfs(1) - sqrt(2*Var) * erfinv(2/(ensN + 1)-1);
+                    % find the mean of the right gaussian tail
+                    muright = xfs(end) - sqrt(2*Var) * erfinv(1-2/(ensN + 1));
+                    mu = [muleft, muright];
                 else
-                    fprintf('Error! Gaussian tail not provided!\n');
+                    fprintf('Error! Only Gaussian Tail is supported as of now!\n');
                     return;
                 end
-                                
-                inn = y(i) - sort(Hxf(i,:));
+
+                % calculate the likelihood for this state
+                innovation = y(i) - sort(Hxf(i, :));
                 ntemp = length(y(i));
-                likelihood = (diag((1/sqrt((2*pi)^ntemp*det(R(i,i)))) * ...
-                    exp(-0.5*inn.'*(R(i,i)\inn)))).';
-                likelihood = likelihood/min(likelihood); % for scaling
-                l_ht = [likelihood(1), 0.5*(likelihood(ind1) + likelihood(ind2)), likelihood(end)];
-                
+                likelihood = (diag((1 / sqrt((2 * pi)^ntemp*R(i, i)))* ...
+                    exp(-0.5*innovation.'*(R(i, i) \ innovation)))).';
+                likelihood = likelihood / min(likelihood); % for scaling
+                likelihoodht = [likelihood(1), 0.5 * (likelihood(ind1) + likelihood(ind2)), likelihood(end)];
```

Comment:
why not just override likelihood instead of creating a new variable

Reply:
Done